### PR TITLE
Blog Reminders: Replace `localSettings` with lookup method calls

### PR DESCRIPTION
**Pull request by crazytonyli**
See #21008 for the reason why we'd want to remove the `localSettings` function, which calls `performQuery` and returns a `NSManagedObject` instance.

I'm not sure what's the best way to test this. I tapped a few blogging prompt features and didn't find any obvious issue.

## Regression Notes
1. Potential unintended areas of impact
None.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
None.

3. What automated tests I added (or what prevented me from doing so)
None.

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

UI Changes testing checklist: N/A

### CODE REVIEW

**Comment by mokagio** [comment]
@crazytonyli I didn't get to review this in time for the 22.8 code freeze. 

I'm going to push the milestone to 22.9. If that doesn't work, feel free to bush back to 22.8 and target the release branch when ready. Sorry for the trouble.

File: WordPress/Classes/ViewRelated/Blog/Blogging Reminders/BloggingRemindersFlowSettingsViewController.swift [hunk 1]
```diff
@@ -684,10 +695,16 @@ private extension BloggingRemindersFlowSettingsViewController {
     ///
     /// - Returns: A closure used to reset changes made to the prompt settings. Returns nil if the update condition is not fulfilled.
     func temporarilyUpdatePromptSettings() -> (() -> Void)? {
+        assert(Thread.isMainThread, "This function must be called from the main thread.")
+
+        // FIXME: Refactor this function to save asynchronously to avoid using the main context.
+        let context = coreDataStack.mainContext
```
Comment (mokagio) [inline WordPress/Classes/ViewRelated/Blog/Blogging Reminders/BloggingRemindersFlowSettingsViewController.swift]: @crazytonyli is this something that should be done as a followup PR?

Comment (crazytonyli) [reply WordPress/Classes/ViewRelated/Blog/Blogging Reminders/BloggingRemindersFlowSettingsViewController.swift]: I could. There are other people looking at the blog reminder area at the moment, but I think a refactor to this function alone shouldn't cause a clash with them.

Comment (crazytonyli) [reply WordPress/Classes/ViewRelated/Blog/Blogging Reminders/BloggingRemindersFlowSettingsViewController.swift]: Addressed in https://github.com/wordpress-mobile/WordPress-iOS/pull/21016/commits/d2ac846f6660dbd5b1d7eae1e9d2cd87aea0e896

**Review by mokagio — APPROVED** [review]

**Pull request merged**